server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2535,47 +2535,44 @@ private void doCheckIndex() throws IOException {
         if (!Lucene.indexExists(store.directory())) {
             return;
         }
-
-        try (
-            BytesStreamOutput os = new BytesStreamOutput();
-            PrintStream out = new PrintStream(os, false, StandardCharsets.UTF_8.name())
-        ) {
-            if ("checksum".equals(checkIndexOnStartup)) {
-                // physical verification only: verify all checksums for the latest commit
-                IOException corrupt = null;
-                MetadataSnapshot metadata = snapshotStoreMetadata();
-                for (Map.Entry<String, StoreFileMetadata> entry : metadata.asMap().entrySet()) {
-                    try {
-                        Store.checkIntegrity(entry.getValue(), store.directory());
-                        out.println("checksum passed: " + entry.getKey());
-                    } catch (IOException exc) {
-                        out.println("checksum failed: " + entry.getKey());
-                        exc.printStackTrace(out);
-                        corrupt = exc;
-                    }
-                }
-                out.flush();
-                if (corrupt != null) {
-                    logger.warn("check index [failure]\n{}", os.bytes().utf8ToString());
-                    throw corrupt;
+        BytesStreamOutput os = new BytesStreamOutput();
+        PrintStream out = new PrintStream(os, false, StandardCharsets.UTF_8.name());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. Currently, there are execution paths that do not contain closure statements, for example, when an if condition at line 2541 evaluates to true, and an else condition at lines 2556, 2574 evaluate to true. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).